### PR TITLE
Hendrik

### PR DESCRIPTION
**Pull request by Hevhove**
Updated latest features with working GET and POST requests, with a basic minimal database in a json file working with php.

### CODE REVIEW

File: includes/utils.hpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 # include <iostream>
 # include <stdio.h>
 # include <fstream>
+# include <fcntl.h>
```
Comment (nprimo) [inline includes/utils.hpp:36]: In this file there should be an `# include <algorithm>`

File: srcs/server/Server.cpp [hunk 2]
```diff
@@ -164,19 +183,26 @@ void	Server::handleNewConnection(void) {
 }
 
 void	Server::dropConnection(int i) {
-	// Remove from the pfds list by copying the file descriptor of the last one over it. 
-    _connections[_pfds[i].fd]->closeSocket();
-    _pfds[i] = _pfds[_fd_count - 1];
-	_fd_count--;
```
Comment (nprimo) [inline srcs/server/Server.cpp:180]: line 186 - in the `catch` statement - you are missing a reference (`&`) to the `std::exception e`

File: srcs/server/Response.cpp [hunk 3]
```diff
@@ -61,11 +53,9 @@ void    Response::setDateHeader(void) {
 }
```
Comment (nprimo) [inline srcs/server/Response.cpp:52]: in line 44 - `nullptr` is a c++11 keyword apparently
image

To be changed with `NULL`?

File: includes/GetResponse.hpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (nprimo) [inline includes/GetResponse.hpp:1]: Why adding a new Class and not working on the existing Response Class?

File: Makefile [hunk 5]
```diff
@@ -1,17 +1,24 @@
 # VARIABLE DEFINITIONS
 NAME			= webserv
+
 SRCS_DIR		= srcs
-SRCS			= $(wildcard srcs/*.cpp)
+SRCS			:= $(shell find $(SRCS_DIR) -name *.cpp)
+
 OBJS_DIR		= objs
 OBJS			:= $(SRCS:$(SRCS_DIR)/%.cpp=$(OBJS_DIR)/%.o)
```
Comment (nprimo) [inline Makefile:8]: to avoid relinking here is the update line:
```
OBJS			:= $($(SRCS_DIR)/%.cpp=$(OBJS_DIR)/%.o)
```

**Review by nprimo — CHANGES_REQUESTED** [review]
To change the Makefile to avoid relinking - I left a comment in line to fix that

File: Makefile (hunk 5 above)
Comment (Hevhove) [reply Makefile:8]: Are you sure of the above? Not working on my system for some reason. If I make any changes, webserv is always up to date and it never compiles anymore. If I fclean first, then run it, it has no input.

Comment (nprimo) [reply Makefile:8]: You're right - the fix I suggested is breaking up the `Makefile`. The issue is linked to the rule `compile_msg` - if you remove it from the objects build up dependencies you should have a functioning `Makefile` that does not recompile

Here are the diffs to a version that does not relink
```
diff --git a/Makefile b/Makefile
index fd440fa..0529262 100644
--- a/Makefile
+++ b/Makefile
@@ -27,7 +27,8 @@ else
 endif
  
 # RULES
-$(OBJS_DIR)/%.o: $(SRCS_DIR)/%.cpp compile_msg
+$(OBJS_DIR)/%.o: $(SRCS_DIR)/%.cpp 
+       @printf "[$(C_YELLOW)=====COMPILING=====$(C_END)]\n"
        @mkdir -p $(OBJS_DIR)
        @mkdir -p $(OBJS_SUB_DIR)
        $(CC) $(CFLAGS) $(INCLUDES) -c $< -o $@
@@ -39,8 +40,6 @@ $(NAME): $(OBJS)
  
 all: $(NAME)
  
-compile_msg:
-       @printf "[$(C_YELLOW)=====COMPILING=====$(C_END)]\n"
  
 del_msg:
        @printf "[$(C_RED)======DELETING=====$(C_END)]\n"
(END)
```

Comment (Hevhove) [reply Makefile:8]: I see, but unfortunately we don't see the header "COMPILING" correctly anymore. Been trying different makefiles just now and nothing seems to work to display this header before giving a list of compilation commands :/

Comment (nprimo) [reply Makefile:8]: I am not a big fan of all this extra information to be honest :) I prefer a clean log eheh 

But I am happy that we finally manage to find the cause of the issue, I guess that's the important thing at the end of the day ;)

**Review by nprimo — APPROVED** [review]
LGTM!

**Comment by Hevhove** [comment]
Added in most recent keep-alive branch in here.

**Pull request merged**